physics.py

```diff
@@ -141,6 +141,32 @@ def update(self, limelightPose: Pose2d, targetPose: Pose2d) -> None:
         )
 
 
+class IntakeCameraSim:
+    def __init__(self) -> None:
+        NetworkTables.initialize()
+        self.photonvisionNetworkTable = NetworkTables.getTable(
+            constants.kPhotonvisionNetworkTableName
+        )
+
+    def update(self, intakeCameraPose: Pose2d, ballPose: Pose2d) -> None:
+        ballInCamera = Transform2d(intakeCameraPose, ballPose)
+        ballAngle = convenientmath.rotationFromTranslation(ballInCamera.translation())
+        ballValid = False
+        if (
+            constants.kIntakeCameraMinHorizontalFOV.radians()
+            < ballAngle.radians()
+            < constants.kIntakeCameraMaxHorizontalFOV.radians()
+        ):
+            ballValid = True
+            ballDistance = intakeCameraPose.translation().distance(
+                ballPose.translation()
+            )
+            self.photonvisionNetworkTable.putNumber(constants.kPhotonvisionTargetSimDistanceKey, ballDistance)
```

Comment:
photonvision doesnt export distance I thought?
you need to export the vertical offset and then within the vision subsystem calculate the actual distance